### PR TITLE
More consistent asJava and asScala methods

### LINKED ISSUE #5722: Consistent asScala and asJava methods on components
All the Scala/Java conversion methods should be named consistently. For example on `RequestHeader` the `_underlyingHeader` one should be called `asScala` and there should be an `asJava` on the Scala one.

### PR DESCRIPTION
**Pull request by marcospereira**
## Fixes

Fixes #5722

## Purpose

This adds some missing conversions between Java and Scala types. This issue is also one of the steps to have Java components.

### CODE REVIEW

File: documentation/manual/working/javaGuide/code/MockJavaAction.scala [hunk 1]
```diff
@@ -54,18 +54,18 @@ object MockJavaActionHelper {
   def call(action: Action[Http.RequestBody], requestBuilder: play.mvc.Http.RequestBuilder)(implicit mat: Materializer): Result = {
     Helpers.await(requestBuilder.body() match {
       case null =>
-        action.apply(requestBuilder.build()._underlyingRequest)
+        action.apply(requestBuilder.build().toScala)
```
Comment (benmccann) [inline documentation/manual/working/javaGuide/code/MockJavaAction.scala]: The PR title says `asScala`

**Comment by marcospereira** [comment]
@gmethvin and @benmccann see here:

https://github.com/playframework/playframework/pull/7163/files#diff-305d0ed8b9666461d89742493845c56bR736

**Comment by gmethvin** [comment]
Can't we just override `asScala` with more specific return type in `Request`?

**Comment by marcospereira** [comment]
Done. This is ready to be reviewed again.

**Pull request merged**